### PR TITLE
Add custom encoder

### PR DESCRIPTION
**Pull request by rauchg**
As discussed in #7.

### CODE REVIEW

File: src/__tests__/index.js [hunk 1]
```diff
@@ -113,6 +113,19 @@ const tests = [{
...
+        '@keyframes whiteToBlack{0%{color:#fff}to{color:#000}}.one{animation: 100ms whiteToBlack}',
+        '@counter-style custom{system:extends decimal;suffix:"> "}ol{list-style:custom}',
+        'body{counter-reset:section}h3:before{counter-increment:section;content:"Section" counter(section) ": "}'
+    ].join(''),
+    expected: [
+        '@keyframes PREFIXwhiteToBlack{0%{color:#fff}to{color:#000}}.one{animation: 100ms PREFIXwhiteToBlack}',
+        '@counter-style PREFIXcustom{system:extends decimal;suffix:"> "}ol{list-style:PREFIXcustom}',
+        'body{counter-reset:PREFIXsection}h3:before{counter-increment:PREFIXsection;content:"Section" counter(PREFIXsection) ": "}'
+    ].join(''),
+    options: {encoder: (i, val) => `PREFIX${val}`}
```
Comment (ben-eb) [inline src/__tests__/index.js]: Do you think you can reverse the arguments? `value, index` instead of `index, value`? It's more consistent with `[].forEach`, `[].map` et al. :smile:

Comment (rauchg) [inline src/__tests__/index.js]: Makes sense.

**Comment by rauchg** [comment]
Updated

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "postcss-reduce-idents",
-  "version": "2.2.2",
+  "name": "postcss-reduce-idents-rauchg",
```
Comment (ben-eb) [inline package.json]: Don't need this here :smile:

**Comment by rauchg** [comment]
Sorry I was trying to publish for my own use.

**Comment by rauchg** [comment]
Reverted

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
...
+  "version": "2.3.0",
```
Comment (ben-eb) [inline package.json]: Can you remove the version bump please? I use `npm version` which automatically commits, increments the version and tags the release. :wink:

Comment (rauchg) [inline package.json]: I did

File: README.md [hunk 4]
```diff
@@ -88,26 +88,37 @@ examples for your environment.
 
 ##### counter
 
-Type: `boolean`
```
Comment (TrySound) [inline README.md]: Why did you remove spaces?

**Comment by TrySound** [comment]
@rauchg Also squash commits please.

File: README.md (hunk 4 above)
Comment (rauchg) [inline README.md]: OCD. I can revert them if you'd like

Comment (ben-eb) [inline README.md]: The spaces there tell GitHub to break the line, it looks better for options documentation. :smile:

Comment (rauchg) [inline README.md]: Gotcha. I'll revert

**Comment by rauchg** [comment]
And there you have it folks. The perfect pull request. 🌊👏✨🔥

**Comment by rauchg** [comment]
@TrySound @ben-eb is there anything else I should tweak? 🙌

**Comment by TrySound** [comment]
@rauchg Everything looks fine. We just need a bit time.

**Comment by ben-eb** [comment]
Thanks! :+1:
